daemon/config/registry.go

```diff
@@ -0,0 +1,63 @@
+package config // import "github.com/docker/docker/daemon/config"
+
+import (
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/docker/docker/registry"
+	"github.com/moby/buildkit/util/resolver"
+	resolverconfig "github.com/moby/buildkit/util/resolver/config"
+)
+
+// RegistryHosts returns registry configuration in containerd resolvers format.
+func (conf *Config) RegistryHosts() docker.RegistryHosts {
```

Comment:
Reviewer's note: this is just moved from `daemon/daemon.go`

Reply:
Sadly, I had to move it back. On a positive note, reloaded registry configuration will now be picked up by Buildkit!